### DISCUSSION TITLE
Case sensitive package names when using thin mode

### DISCUSSION THREAD

**Discussion by tmpjr** [oracle/python-oracledb#355] [General]
I spent several hours troubleshooting why a callfunc kept failing on me with an error that the function did not exist: 
ORA-06550 identifier 'package_name.function_name' must be declared". 

It worked fine in thick mode not when trying thin. I took a shot in the dark and changed the package name to all upper case. So, I went from calling "package_name.function" to "PACKAGE_NAME.function" and that fixed the issue. I did not see this documented on https://python-oracledb.readthedocs.io/en/latest/user_guide/appendix_b.html but thought I'd bring it up here as a discussion point. Since I'm using this library in AWS lambda I'd rather use thin mode to keep my deploy size down. Are there any other things I might ru

**Comment by anthony-tuininga** [comment]
Can you supply a simple test case that demonstrates the problem? What you are describing doesn't make sense, so I'd like to know what is going on behind the scenes to make it happen! Thanks.

**Reply by anthony-tuininga** [reply]
If you are able to run the test suite (or just the test_3200_features_12_1.py) that might help determine if it is specific to your database configuration or that specific package. It might also help to know the name of the package.

**Reply by tmpjr** [reply]
I suspect it's our db configuration. It's a legacy database that's been around since 2008. It's on 19c now but a lot of old config settings that we're working on updating. One thing we did change recently was sec_case_sensitive_logon = True but not sure why that would affect package name. I'll try to post some code later or tomorrow.

**Comment by tmpjr** [comment]
This morning, I destroyed my CDK stack for the lambda I was using to test. Redeployed and re-typed the package name in mixed case just to be cheeky.. and it worked fine. I'll chalk that up to fatigue from working on legacy code for too long!
